src/Voron/Platform/Win32/WindowsMemoryMapPager.cs

```diff
@@ -132,10 +152,49 @@ public struct SplitValue
             }
 
             NumberOfAllocatedPages = _totalAllocationSize / Constants.Storage.PageSize;
+            long numberOfAllocatedSegments = (_totalAllocationSize / PrefetchSegmentSize) + 1;
+            this._prefetchTable = new byte[(numberOfAllocatedSegments / 2) + 1];
 
             SetPagerState(CreatePagerState());
         }
 
+        public override byte* AcquirePagePointer(IPagerLevelTransactionState tx, long pageNumber, PagerState pagerState = null)
+        {
+            // We need to decide what pager we are going to use right now or risk inconsistencies when performing prefetches from disk.
+            var state = pagerState ?? _pagerState;
+
+            if (PlatformDetails.CanPrefetch)
```

Comment:
Isn't this really eager operation?

Reply:
From the call chain, there is really no reason why to delay that. At that point the call-stack is already in control of the pager, so noone is gonna change that either way.